recipe/meta.yaml

```
@@ -42,27 +42,41 @@ requirements:
     - python >3               # [not osx]
 
 outputs:
-  - name: libcxx-devel
+  # This output is generic so that we can use the headers for cross compiling
+  - name: libcxx-headers
     build:
-      string: h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}        # [hardening == "none"]
-      string: debug_h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}  # [hardening == "debug"]
-      run_exports:                                      # [hardening == "debug"]
-        # packages built with hardened lib must not be installable without extra label
-        - libcxx =*=debug*                              # [hardening == "debug"]
+      noarch: generic
     files:
       - include/c++                 # [unix]
       # for C++20 modules, which are installed into $PREFIX/share/libc++
       - share/libc++                # [unix]
       - Library/include/c++         # [win]
       # also for C++20 modules
       - lib/libc++.modules.json     # [unix]
+    requirements:
+      run:
+        - __unix   # [unix]
+        - __win    # [win]
+
+  - name: libcxx-devel
+    build:
+      string: h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}        # [hardening == "none"]
+      string: debug_h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}  # [hardening == "debug"]
+      run_exports:                                      # [hardening == "debug"]
+        # packages built with hardened lib must not be installable without extra label
+        - libcxx =*=debug*                              # [hardening == "debug"]
+    files:
+      # Add non-existent path as this is a meta package
+      - include2/c++
     requirements:
       host:
         - {{ pin_subpackage("libcxx", exact=True) }}
+        - {{ pin_subpackage("libcxx-headers", max_pin="x.x.x") }}
```

I think it'd be cleaner with
```suggestion
        - {{ pin_subpackage("libcxx-headers", exact=True) }}
```
at build time; at runtime we can probably loosen it a bit (probably `'x.x'` would be enough, but fine with `'x.x.x'`).